### ISSUE TITLE
`PrivMsg::is_moderator` returns false with tag `"mod:" "1"`

### ISSUE THREAD

**Issue by Zenithsiz** [museun/twitchchat#161]
Currently `PrivMsg::is_moderator` simply gets the tag `mod` as a bool, but the requests I'm received from twitch give the `mod` field either a `"0"` or `"1"`. I'm not sure if this is a recent change or something only happening in certain channels, but I believe it should be supported either way, as it's a possible response from twitch.

The problem is that `str::parse::<bool>` doesn't parse `"1"` as true, instead giving an `Err`.

A suggested fix would be changing `Tags::get_as_bool` to support `"1"` and `"0"` with something like:
```rust
pub fn get_as_bool<K: ?Sized>(&self, key: &K) -> bool
where
    K: Borrow<str>,
{
    self.get_parsed(key).unwrap_or_else(|| self.get_parsed::<K, u8>(key).map(|num| num == 1))
}
```
or with matching:
```rust
pub fn get_as_bool<K: ?Sized>(&self, key: &K) -> bool
where
    K: Borrow<str>,
{
	match self.get_parsed::<K, bool>(key) {
		Some(b) => b,
		None => self.get_parsed::<K, u8>(key) {
			Some(1) => true,
			_ => false,
		}
	}
}
```

I don't have means of checking if either of them would compile successfully currently, but I assume they should.

**Comment by museun** [comment]
Ah. I've fixed this locally.

```rust
        match self.get_ref(key) {
            Some(d) if &*d == "1" => true,
            Some(d) if &*d == "0" => false,
            Some(d) => d.parse().ok().unwrap_or_default(),
            None => false,
        }
```

Which handles 1, 0, true, false. Does Twitch send 'truthy' values as `1`/`0`? I thought it was `true`/`false`

(note, this code may not reflect the current api. I'm working on making the tags lazy and to reduce the struct size by 32~ bytes.)

**Comment by museun** [comment]
I've published v0.11.7 with these changes. Thanks for letting me know.

**Comment by Zenithsiz** [comment]
I've only started using this library relatively recently, so perhaps before twitch sent `true`/`false`.

Thanks for the quick fix, if anyone needs a quick workaround until this reaches stable, I've been using:
```rust
msg.tags.get("mod").as_deref() == Some("1")
```
Although this does assume that twitch no longer sends `true` / `false` values.
